src/App.jsx

```diff
@@ -1,12 +1,195 @@
+    
+    setIsSearching(true);
+    setMovieList([]);
+    setSearchTerm(searchInput); //set search term state so that when user wants to load more search results the query persists across renders
+    setPageNum(1); //lastly set the Pagenum to 1 to trigger the useEffect ***when pageNum in now playing is > 1*** If user never paginated this wont trigger a use effect. What will trigger it is the change of the searchTerm
+  }
+
+  //function actually calls api and that returns an array of movies that match the searchTerm
+  async function getSearchResults(searchTerm, pageIdx){
+
+    console.log("in get search results");
+    const urlWithQuery = `${baseSearchUrl}?query=${searchTerm}&include_adult=false&language=en-US&page=${pageIdx}`;
+
+    try{
+
+      const res = await fetch(urlWithQuery, {
+        headers:{
+          accept: 'application/json',
+          Authorization: `bearer ${import.meta.env.VITE_API_KEY}`,
+        },
+      });
+      if(!res.ok){
+        throw new Error('Bad api request');
+      }
+      
+      const moviesFound = await res.json();
+      const searchResults = moviesFound.results; //gives us the actual movieFound array
+
+      setMovieList( (prevSearchResults) => [...prevSearchResults, ...searchResults]); //by setting MovieList to be the concatentation of the prev list and the new results we keep tacking to search results if usre clicks load more.
+
+    } catch(err){
+
+      console.log("No search results");
+      console.log(err);
+    }
+
+  }
+
+  //function will fire when user hits the clear button
+  function handleClear(clearInput){
+    setIsSearching(false); //just change isSearching flag to false to when useEffect goes off we just re-render page with now playing movies
+    setMovieList([]); //remember to make MovieList empty again if not you are just goign to append the now playing movies to the end of previous search results
+    setSearchTerm(clearInput); //make search term an empty string to trigger useEffect
+    setPageNum(1); //make sure page is reset for pagination.
+  }
+
+  useEffect(() => { //useEffect is observing the pageNum and searchTerm state. When load more is clicked and page num changes OR we get a new searchTerm itll execute
+    if (isSearching && searchTerm) {
+      getSearchResults(searchTerm, pageNum);
+    } else {
+      getMovieList(pageNum);
+    }
+  }, [pageNum, searchTerm])
+
+
+  //this function will be passed down to SortDropdown component so that in that component we can determine which sort user clicks and call this function and sent result back up to parent in the form of "sortType". All sorting logic and updating of movieList happens here so we dont have to pass all that down
+  function handleSort(sortType){
+
+    //if user clicks back to default option (unsorted) make sure to clear array of sorted one first and then populate now playing
+    if(sortType === 'default'){
+      setMovieList([]);
+      getMovieList(pageNum);
```

Comment:
Is this working fine? Not sure if is because is an async function, but usually when you update a state variable and try to use it right after that (in this case inside the function), you can have race conditions as the variable could not be updated yet. To avoid that, you can decouple that logic on your function, having the function that fetch the movie list based on some parameters as you have it now, but letting the caller decide what to do with that information, for example:
```suggestion
      getMovieList(pageNum).then(movieList => setMovieList(movieList));
```